From 92ceb6c30b4c111ae3e166ce97bdd4fcec9dc5e5 Mon Sep 17 00:00:00 2001
From: EternalMikaelson <klaus_mikaelson@rediffmail.com>
Date: Fri, 9 Oct 2026 01:04:08 +0000
Subject: [PATCH 4/4] camera: Strip OppoGallery dependence in thumbnail-tap launch

GalleryHelper.m() used an explicit-package/resolveActivity() launch
path with an OEM-specific error dialog as fallback, which fires
whenever OppoGallery2's manifest intent-filters don't resolve the
way the OEM stack expects on AOSP. Replaced with a plain implicit
ACTION_VIEW intent instead, letting Android's normal default-app
resolution handle it.

Change-Id: I82d0117294cda2020a9d725da1e29ee6b936fdc7
Signed-off-by: EternalMikaelson <klaus_mikaelson@rediffmail.com>
---
 .../oplus/camera/helper/GalleryHelper.smali   | 141 ++----------------
 1 file changed, 15 insertions(+), 126 deletions(-)

diff --git a/smali_classes12/com/oplus/camera/helper/GalleryHelper.smali b/smali_classes12/com/oplus/camera/helper/GalleryHelper.smali
index 3a0177cf..d2b017d2 100644
--- a/smali_classes12/com/oplus/camera/helper/GalleryHelper.smali
+++ b/smali_classes12/com/oplus/camera/helper/GalleryHelper.smali
@@ -2311,145 +2311,34 @@
 .end method
 
 .method public final m(Landroid/content/Intent;ZLandroid/net/Uri;)V
-    .locals 1
-
-    .line 1
-    new-instance v0, Loe/b;
-
-    .line 2
-    .line 3
-    invoke-direct {v0, p2, p3}, Loe/b;-><init>(ZLandroid/net/Uri;)V
-
-    .line 4
-    .line 5
-    .line 6
-    const-string p2, "thumbnail_click"
-
-    .line 7
-    .line 8
-    invoke-static {p2, v0}, Lcom/oplus/camera/statistics/ReportThread;->postEvent(Ljava/lang/String;Ljava/lang/Runnable;)V
-
-    .line 9
-    .line 10
-    .line 11
-    new-instance p2, Ljava/lang/ref/WeakReference;
-
-    .line 12
-    .line 13
-    invoke-direct {p2, p0}, Ljava/lang/ref/WeakReference;-><init>(Ljava/lang/Object;)V
-
-    .line 14
-    .line 15
-    .line 16
-    sput-object p2, Lcom/oplus/camera/helper/GalleryHelper;->B:Ljava/lang/ref/WeakReference;
-
-    .line 17
-    .line 18
-    iget-object p2, p0, Lcom/oplus/camera/helper/GalleryHelper;->a:Landroid/app/Activity;
-
-    .line 19
-    .line 20
-    invoke-virtual {p2}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
-
-    .line 21
-    .line 22
-    .line 23
-    move-result-object p2
-
-    .line 24
-    invoke-virtual {p1, p2}, Landroid/content/Intent;->resolveActivity(Landroid/content/pm/PackageManager;)Landroid/content/ComponentName;
-
-    .line 25
-    .line 26
-    .line 27
-    move-result-object p2
-
-    .line 28
-    if-eqz p2, :cond_0
-
-    .line 29
-    .line 30
-    invoke-static {}, Landroid/app/ActivityOptions;->makeBasic()Landroid/app/ActivityOptions;
-
-    .line 31
-    .line 32
-    .line 33
-    move-result-object p2
-
-    .line 34
-    iget-object v0, p0, Lcom/oplus/camera/helper/GalleryHelper;->a:Landroid/app/Activity;
-
-    .line 35
-    .line 36
-    invoke-virtual {v0}, Landroid/content/Context;->getDisplay()Landroid/view/Display;
-
-    .line 37
-    .line 38
-    .line 39
-    move-result-object v0
-
-    .line 40
-    invoke-virtual {v0}, Landroid/view/Display;->getDisplayId()I
-
-    .line 41
-    .line 42
-    .line 43
-    move-result v0
+    .locals 2
 
-    .line 44
-    invoke-virtual {p2, v0}, Landroid/app/ActivityOptions;->setLaunchDisplayId(I)Landroid/app/ActivityOptions;
+    new-instance v0, Landroid/content/Intent;
 
-    .line 45
-    .line 46
-    .line 47
-    invoke-static {p1, p3}, Lcom/oplus/camera/helper/GalleryHelper;->a(Landroid/content/Intent;Landroid/net/Uri;)V
+    const-string v1, "android.intent.action.VIEW"
 
-    .line 48
-    .line 49
-    .line 50
-    iget-object p0, p0, Lcom/oplus/camera/helper/GalleryHelper;->a:Landroid/app/Activity;
+    invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
 
-    .line 51
-    .line 52
-    invoke-virtual {p2}, Landroid/app/ActivityOptions;->toBundle()Landroid/os/Bundle;
+    if-eqz p2, :cond_image
 
-    .line 53
-    .line 54
-    .line 55
-    move-result-object p2
+    const-string v1, "video/*"
 
-    .line 56
-    invoke-virtual {p0, p1, p2}, Landroid/app/Activity;->startActivity(Landroid/content/Intent;Landroid/os/Bundle;)V
+    goto :goto_setdata
 
-    .line 57
-    .line 58
-    .line 59
-    goto :goto_0
+    :cond_image
+    const-string v1, "image/*"
 
-    .line 60
-    :cond_0
-    new-instance p0, Leh/d;
+    :goto_setdata
+    invoke-virtual {v0, p3, v1}, Landroid/content/Intent;->setDataAndType(Landroid/net/Uri;Ljava/lang/String;)Landroid/content/Intent;
 
-    .line 61
-    .line 62
-    const/4 p1, 0x0
+    const/4 v1, 0x1
 
-    .line 63
-    invoke-direct {p0, p1}, Leh/d;-><init>(I)V
+    invoke-virtual {v0, v1}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
 
-    .line 64
-    .line 65
-    .line 66
-    const-string p1, "GalleryHelper"
+    iget-object v1, p0, Lcom/oplus/camera/helper/GalleryHelper;->a:Landroid/app/Activity;
 
-    .line 67
-    .line 68
-    invoke-static {p1, p0}, Lw6/c;->z(Ljava/lang/String;Ljava/util/function/Supplier;)V
+    invoke-virtual {v1, v0}, Landroid/app/Activity;->startActivity(Landroid/content/Intent;)V
 
-    .line 69
-    .line 70
-    .line 71
-    :goto_0
     return-void
 .end method
 
-- 
2.43.0

